addons/pos_self_order/models/pos_order.py

```diff
@@ -11,6 +11,18 @@ class PosOrderLine(models.Model):
 
     # For the moment we need this to keep attributes consistency between the server and client_side.
     selected_attributes = fields.Json(string="Selected Attributes")
+    combo_parent_id = fields.Many2one('pos.order.line', string='Combo Parent')
+    combo_line_ids = fields.One2many('pos.order.line', 'combo_parent_id', string='Combo Lines')
+    combo_id = fields.Many2one('pos.combo', string='Combo line reference')
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if (vals.get('combo_parent_id')):
+                vals.update([
+                    ('combo_parent_id', self.search([('uuid', '=', vals.get('combo_parent_id'))]).id)
```

Comment:
combo_parent_id is an id or a uuid?

Reply:
@adgu-odoo can you look at this one ?

Reply:
I've changed the notation of the variables. The frontend will give the uuid of the parent line to the backend. The backend will then find the correct line using the uuid and add the field combo_parent_id to the line before creating it. It must be done this way as the parent line and the child line are sent to the backend at the same time so the server id of the parent is not known at that moment by the children. After that, the backend will first create the parent line and then the child lines.